lib/services/product-manager.js

```diff
@@ -202,6 +203,65 @@ export default class ProductManager {
       : Promise.resolve(product)
   }
 
+  /**
+   * Method will take a product with variants which should be anonymized.
+   * It will modify slug, key and name so the product can be created without
+   * issues with duplicate fields
+   * @param product productDraft
+   */
+  getAnonymizedProduct (product) {
```

Comment:
What if you call this method for 2 products at the same time? Then they both will have the same timestamp => same slug and key? Maybe we should add some additional random element (e.g. append with a random number from 0..100) to eliminate this problem. And don't forget to write a test :)

Reply:
We are using now unix timestamp in milliseconds so there would have to be running two variant reassignments in parallel and they would have to call this function exactly at the same time which is very unlikely.

Also, this code is synchronous and because of the node.js single threaded model even if you call this function two times it won't be processed at the same time. Let me add a test for it.

Reply:
Test added here d23aec3 - even if you call this function two times, according to a timestamp, it takes approx 4 milliseconds to anonymize product so the slugs are slightly different

Reply:
So you think even if for sth like:
```js
Promise.all([productService.getAnonymizedProduct(productDraft), productService.getAnonymizedProduct(productDraft)])
```
the two slugs will be different? Your test doesn't reflect that.

Reply:
Also, the name should be `getAnonymizedProductDraft`? If it's a product, then the slug is in `masterData.current.slug` or `masterData.staged.slug`. In this project, we use both product and productDraft/productProjection, so we should name it more specific.

Reply:
I updated the method name + added additional salt which should make the key unique c63c7ae (plus changed tests).

I tried that Promise.all and it sometimes returned the same timestamp - it is not because it runs in parallel (node.js is single threaded so no two synchronous functions can run at the same time) but both functions were processed at the same millisecond so the timestamp was the same. Thx for catching it.